mne/annotations.py

```diff
@@ -102,6 +103,17 @@ def __len__(self):
         """Return the number of annotations."""
         return len(self.duration)
 
+    def __add__(self, other):
+        """Add (concatencate) two Annotation objects."""
+        new = deepcopy(self)
```

Comment:
How about we instead 1) a `copy` method that does `return deepcopy(self)`, and 2) make `append` return `self` for chaining, then this can be:

```
return self.copy().apppend(...)
```
and the one below can be
```
return self.append(...)
```

Reply:
this is already an append method that takes onset, duration and description.

if you think of it as a list + makes more sense than append.

anyway you need tests

Reply:
I like Eric's proposal.

Reply:
@larsoner what's the benefit of doing it like you suggest? Couldn't it be problematic that `append` both returns `self` and modifies it in-place? Lists don't work like this, so this might be confusing.

Reply:
I actually think chaining is more useful than the default list behavior, I'm usually more surprised if objects don't allow for chaining. And many mne objects follow this pattern - for example methods of `raw` operate in-place and return `self` for chaining.

Reply:
OK, but then we don't really need to define the operators, do we? If `append` returns `self`, we can add two `Annotations` object ourselves with the `copy` and `append` methods. Not sure if this is more intuitive though.

Reply:
Plus, this PR is about adding `+` and `+=` to the public API, and changing the `append` method would unnecessarily change this public method. However, if we *want* to change `append` I'm certainly happy to implement the operators this way.

Reply:
yes but we already have an append right?

Reply:
Yes, but you have to pass three arguments (`np.array`) instead of one `Annotations` object.